samcli/hook_packages/terraform/hooks/prepare/resources/apigw.py

```diff
@@ -91,3 +106,105 @@ class ApiGatewayStageProperties(ResourceProperties):
+      Properties:
+        Integration:
+          Uri: Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${function.Arn}/invocations
+          Type: AWS_PROXY
+
+    Parameters
+    ----------
+    cfn_dict: dict
+        Resultant CFN dict that will be mutated to append integrations to each API Gateway method
+    gateway_integrations_cfn: Dict[str, List]
+        Dict containing Internal API Gateway integrations to be appended to the CFN dict
+    """
+    for logical_id, resource in cfn_dict.get("Resources").items():
+        if resource.get("Type", "") == AWS_APIGATEWAY_METHOD:
+            resource_properties = resource.get("Properties", {})
+            search_key = _gateway_method_integration_identifier(resource_properties)
+            integration_properties = _find_gateway_integration(search_key, gateway_integrations_cfn)
+            if not integration_properties:
+                LOG.debug("A corresponding gateway integration for the gateway method %s was not found", logical_id)
+                continue
+            _create_gateway_method_integration(resource, integration_properties)
+
+
+def _find_gateway_integration(search_key: set, gateway_integrations_cfn: Dict[str, List]) -> Optional[dict]:
+    """
+    Iterate through all internal API Gateway integrations and search of an
+    integration whose unique identifier matches the given search key.
+
+    Parameters
+    ----------
+    search_key: set
+        Set containing the unique identifier of the API Gateway integration to match
+    gateway_integrations_cfn: Dict[str, List]
+        Dict containing all Internal API Gateway integration resources to search through
+
+    Returns
+    -------
+        Properties of the internal API Gateway integration if found, otherwise returns None
+
+    """
+    for _, gateway_integrations in gateway_integrations_cfn.items():
+        for resource in gateway_integrations:
+            resource_properties = resource.get("Properties", {})
+            integration_key = _gateway_method_integration_identifier(resource_properties)
+            if integration_key == search_key:
+                return resource_properties
+    return None
+
+
+def _gateway_method_integration_identifier(resource_properties: dict) -> set:
+    """
+    Given a dict containing the properties that uniquely identify an
+    API Gateway integration (RestApiId, ResourceId, HttpMethod)
+    returns a set containing these fields that be used to check for equality of integrations.
+
+    Parameters
+    ----------
+    resource_properties: dict
+        Dict containing the resource properties that can be used to uniquely identify and API Gateway integration
+
+    Returns
+    -------
+        Returns a set comprised of unique identifiers of an API Gateway integration
+
+    """
+    return {
+        resource_properties.get("RestApiId", {}).get("Ref", ""),
+        resource_properties.get("ResourceId", {}).get("Ref", ""),
```

Comment:
we have a case that these values can be strings if customer used static values for Apigateway that is defined outside the TF project.